lib/gui/lvgl/lvgl.c

```diff
@@ -41,45 +46,33 @@ static uint8_t buf1[BUFFER_SIZE] __aligned(4);
 #endif /* CONFIG_LVGL_BUFFER_ALLOC_STATIC */
 
 #if CONFIG_LVGL_LOG_LEVEL != 0
-static void lvgl_log(lv_log_level_t level, const char *file, uint32_t line,
-		const char *func, const char *dsc)
+/*
+ * In LVGLv8 the signature of the logging callback has changes and it no longer
+ * takes the log level as an integer argument. Instead, the log level is now
+ * already part of the buffer passed to the logging callback. It's not optimal
+ * but we need to live with it and parse the buffer manually to determine the
+ * level and then truncate the string we actually pass to the logging framework.
+ */
+static void lvgl_log(const char *buf)
 {
-	/* Convert LVGL log level to Zephyr log level
-	 *
-	 * LVGL log level mapping:
-	 * * LV_LOG_LEVEL_TRACE 0
-	 * * LV_LOG_LEVEL_INFO  1
-	 * * LV_LOG_LEVEL_WARN  2
-	 * * LV_LOG_LEVEL_ERROR 3
-	 * * LV_LOG_LEVEL_NUM  4
-	 *
-	 * Zephyr log level mapping:
-	 * *  LOG_LEVEL_NONE 0
-	 * * LOG_LEVEL_ERR 1
-	 * * LOG_LEVEL_WRN 2
-	 * * LOG_LEVEL_INF 3
-	 * * LOG_LEVEL_DBG 4
+	/*
+	 * This is ugly and should be done in a loop or something but as it
+	 * turned out, Z_LOG()'s first argument (that specifies the log level)
+	 * cannot be an l-value...
 	 */
-	char *dupdsc = log_strdup(dsc);
-
-	ARG_UNUSED(file);
-	ARG_UNUSED(line);
-	ARG_UNUSED(func);
-
-	switch (level) {
-	case LV_LOG_LEVEL_TRACE:
-		Z_LOG(LOG_LEVEL_DBG, "%s", dupdsc);
-		break;
-	case LV_LOG_LEVEL_INFO:
-		Z_LOG(LOG_LEVEL_INF, "%s", dupdsc);
-		break;
-	case LV_LOG_LEVEL_WARN:
-		Z_LOG(LOG_LEVEL_WRN, "%s", dupdsc);
-		break;
-	case LV_LOG_LEVEL_ERROR:
-		Z_LOG(LOG_LEVEL_ERR, "%s", dupdsc);
-		break;
-	}
+	static const char *const err_str = "[Error]";
+	static const char *const wrn_str = "[Warn]";
+	static const char *const inf_str = "[Info]";
+	static const char *const dbg_str = "[Trace]";
+
+	if (strncmp(buf, err_str, strlen(err_str)) == 0)
+		LOG_ERR("%s", log_strdup(buf + strlen(err_str) + 1));
+	else if (strncmp(buf, wrn_str, strlen(wrn_str)) == 0)
+		LOG_WRN("%s", log_strdup(buf + strlen(wrn_str) + 1));
+	else if (strncmp(buf, inf_str, strlen(inf_str)) == 0)
+		LOG_INF("%s", log_strdup(buf + strlen(inf_str) + 1));
+	else if (strncmp(buf, dbg_str, strlen(dbg_str)) == 0)
+		LOG_DBG("%s", log_strdup(buf + strlen(dbg_str) + 1));
```

Comment:
nit: maybe a faster way would be to just compare against `buf[1]` since it seems to be a set: `E, W, I, T`.